cmd/adhoc/cmd/root.go

```diff
@@ -41,45 +49,13 @@ func onPersistentPreRun(cmd *cobra.Command, args []string) error {
 			return log.Configure(parser)
 		},
 		func() error {
-			// Setting up file formatting
-			viper.SetConfigFile("config/syncron.yaml")
-			viper.SetConfigType("yaml")
-			viper.AddConfigPath(".")
-			// Reading from file
-			err := viper.ReadInConfig()
-			if err != nil {
-				return err
-			} else {
-				logrus.Info("Your configuration file was read succesfully")
-				logrus.Info("Reading from bucket: ", viper.Get("bucket"))
-			}
-			return nil
-		},
-		func() error {
-			// Initialize a session with AWS SDK
-			// It will read from the file located at ~/.aws/credentials and syncron/config/syncron.yaml
-			logrus.Info("Starting setup for AWS s3")
-			sess, err := session.NewSession(&aws.Config{
-				Region:   aws.String(viper.GetString("s3.region")),
-				Endpoint: aws.String(viper.GetString("s3.endpoint")),
-			},
-			)
-			if err != nil {
-				fmt.Println("Error setting up AWS...", "Check config params")
-				return err
-			} else {
-				logrus.Info("AWS setup succesful!")
-			}
-			logrus.Info("Accessing bucket...")
-			svc := s3.New(sess)
-			resp, err := svc.ListObjectsV2(&s3.ListObjectsV2Input{Bucket: aws.String(viper.GetString("bucket"))})
-			if err != nil {
-				fmt.Println("Unable to list items in bucket", resp)
-				return err
-			} else {
-				logrus.Info("Success getting into the bucket!")
-			}
-			return nil
+			// Reading configuration file
+			s3setup.ConfigRead()
+			// Creating AWS session
+			sess := s3setup.SetupSession()
+			// Accessing bucket
+			err := s3setup.AccessBucket(sess)
+			return err
```

Comment:
Additionally, the session is setup but is not accessible to the 'download' command. I believe you will have to move all this to 'download.go'.